nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/AttributesToCSV.java

```diff
@@ -0,0 +1,280 @@
+                    if (val != null && !val.isEmpty()) {
+                        result.put(coreAttribute, val);
+                    } else {
+                        if (nullValForEmptyString) {
+                            result.put(coreAttribute, "null");
+                        } else {
+                            result.put(coreAttribute, "");
+                        }
+                    }
+                }
+            }
+        } else {
+            //remove core attributes since the user does not want them, unless they are in the attribute list. Attribute List always wins
+            for (String coreAttribute : coreAttributes) {
+                //never override user specified attributes, even if the user has selected to exclude core attributes
+                if(!userSpecifiedAttributes.contains(coreAttribute)) {
+                    result.remove(coreAttribute);
+                }
+            }
+        }
+        return result;
+    }
+
+    private Set<String> attributeListStringToSet(String attributeList) {
+        //take the user specified attribute list string and convert to list of strings.
+        Set<String> result = new HashSet<>();
+        if (StringUtils.isNotBlank(attributeList)) {
+            String[] ats = StringUtils.split(attributeList, OUTPUT_SEPARATOR);
+            if (ats != null) {
+                for (String str : ats) {
+                    String trim = str.trim();
+                        result.add(trim);
+                }
+            }
+        }
+        return result;
+    }
+
+    @OnScheduled
+    public void onScheduled(ProcessContext context) {
+        includeCoreAttributes = context.getProperty(INCLUDE_CORE_ATTRIBUTES).asBoolean();
+        coreAttributes = Arrays.stream(CoreAttributes.values()).map(CoreAttributes::key).collect(Collectors.toSet());
+        userSpecifiedAttributes = attributeListStringToSet(context.getProperty(ATTRIBUTES_LIST).getValue());
+        destinationContent = OUTPUT_OVERWRITE_CONTENT.equals(context.getProperty(DESTINATION).getValue());
+        nullValForEmptyString = context.getProperty(NULL_VALUE_FOR_EMPTY_STRING).asBoolean();
+     }
+
+    @Override
+    public void onTrigger(ProcessContext context, ProcessSession session) throws ProcessException {
+        final FlowFile original = session.get();
+        if (original == null) {
+            return;
+        }
+
+        final Map<String, String> atrList = buildAttributesMapForFlowFile(original, userSpecifiedAttributes);
+
+        //escape attribute values
+        final StringBuilder sb = new StringBuilder();
+        for (final String val : atrList.values()) {
+            sb.append(StringEscapeUtils.escapeCsv(val));
+            sb.append(OUTPUT_SEPARATOR);
+        }
+
+        //check if the output separator is at the end of the string, if so then remove it
+        if(sb.length() > 0 && sb.lastIndexOf(OUTPUT_SEPARATOR) == sb.length() -1) {
```

Comment:
I think you could eliminate this by adding `int index = 0;` after the declaration of `sb` and then doing:

`sb.append(index++ < atrList.values().size() -1 ? OUTPUT_SEPARATOR : "")`